tools/java/org.hl7.fhir.tools.core/src/org/hl7/fhir/tools/implementations/go/GoGenerator.java

```diff
@@ -287,6 +290,53 @@ public int compare(ResourceSearchInfo a, ResourceSearchInfo b) {
         controllerWriter.close();
     }
 
+    private void generateIndexesConfig(Definitions definitions, String outputDir, STGroup templateGroup) throws IOException {
+        ST utilTemplate = templateGroup.getInstanceOf("indexes.conf");        
+
+        // Identify for each resource all elements of type REFERENCE
+        ArrayList<ResourceIndexInfo> indexInfos = new ArrayList<ResourceIndexInfo>(definitions.getResources().size());
+        for (ResourceDefn resource : definitions.getResources().values()) {
+
+            ResourceIndexInfo indexInfo = new ResourceIndexInfo(Utilities.pluralizeMe(resource.getName().toLowerCase()));
+
+            for (ElementDefn el : resource.getRoot().getElements()) {
+
+                for (TypeRef typeRef : el.getTypes()) {
+
+                    // check if the element is a reference type
+                    String elementType = typeRef.getName();
+                    if (elementType.equals("Reference")) {
+                        
+                        StringBuilder elementNameBuilder = new StringBuilder(el.getName().replace("[x]", ""));
+                        if (el.getTypes().size() > 1) {
+                            elementNameBuilder.append(Utilities.capitalize(typeRef.getName()));
+                        }
+                        String elementName = elementNameBuilder.toString();
+
+                        MgoIndex index = new MgoIndex(elementName);
+                        indexInfo.addIndex(index);
+                    }
+                } 
+            }
+            indexInfos.add(indexInfo);
+        }
```

Comment:
Unfortunately, this approach isn't the best approach to determining the reference indexes we need.  There are two issues with it: (1) It may created indexes on references that aren't even defined as search parameters -- so they'll never be used, (2) it only goes one level deep on the resource -- it doesn't find any _nested_ references.

There are actually quite a few examples of the second point, but here's one: CarePlan has a search parameter called `performer` with the path `CarePlan.activity.detail.performer` ([spec](http://hl7.org/fhir/2016Sep/careplan.html#search)).  This reference is currently _not_ being indexed since the code doesn't look into nested paths.

I think the right approach is to loop through the search parameters and create indexes for all of the search parameters of type `reference`.  This should address both limitations above.